### PR TITLE
CP charts endpoint

### PR DESCRIPTION
**Pull request by Aeilert**
Hi @tonyfujs ,

Here is a suggested endpoint for all the CP - charts (per country and poverty line). 

Maybe take and extra look at : 

* Selection of countries for Poverty MRV chart (see `cp_pov_mrv_select_countries()` and `test-ui_functions.R`) 
* Selection of year range for Poverty MRV chart 
* Whether we should have a single endpoint or mulitiple endpoints for the charts

### CODE REVIEW

File: R/ui_functions.R [hunk 1]
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
...
+
+  # Fetch data for poverty bar chart
+  region <-
+    lkup$svy_lkup[country_code == country]$pcn_region_code %>%
+    unique()
+  countries <-
+    lkup$svy_lkup[pcn_region_code == region]$country_code %>%
+    unique()
+
+  res_pov_mrv <- pip(country = countries, lkup = lkup)
```
Comment (tonyfujs) [inline R/ui_functions.R:247]: Shouldn't this be:
`res_pov_mrv <- pip(country = countries, povline = povline, lkup = lkup)` Otherwise it will always use the `povline` default value and won't be sensitive to the poverty line chosen by the end-user.

File: R/ui_functions.R [hunk 2]
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
...
+  # If the number of countries in the region of the selected country is > 12,
+  # then the countries to be displayed will be selected as follow:
+  if (nrow(dt) > 12) {
+    h <- dt[country_code == country]$headcount
+    v <- sort(dt$headcount)
+    # IF selected country does not pertain to top 5 or bottom 5, display:
+    if (!(h %in% utils::tail(v, 5) | h %in% utils::head(v, 5))) {
+      v <- v[!v %in% h]
+      v2 <- v[v > h]
+      v3 <- v[v < h]
```
Comment (tonyfujs) [inline R/ui_functions.R:289]: What do you think of replacing  
* `v2` with `v_above`, and
* `v3` with `v_below`

File: R/ui_functions.R [hunk 3]
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
...
+      vals <- c(h, utils::head(v, 3), utils::tail(v, 3),
+                utils::head(v2, 2), utils::tail(v3, 2))
+      # ELSE IF selected country pertains to top 5, display:
+    } else if (h %in% utils::tail(v, 5)) {
+      vals <- c(utils::head(v, 5), utils::tail(v, 6))
+      # ELSE display
+    } else {
+      vals <- c(utils::head(v, 6), utils::tail(v, 5))
+    }
```
Comment (tonyfujs) [inline R/ui_functions.R:298]: I think I would probably extract this whole section into its own function. This is the heart of the selection algorithm, and it would be much easier to unit test edge cases.

**Review by tonyfujs — CHANGES_REQUESTED** [review]
Hey @Aeilert 
Thanks for this. I added a few comments. Take a look and let me know what you think.

File: R/ui_functions.R [hunk 4]
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
...
+#' @param pop_units numeric: Units used to express population numbers (default to million)
+#' @param year_range numeric: Range used to subset countries in Poverty MRV chart
+#' @param lkup list: A list of lkup tables
+#'
+#' @return list
+#' @export
+#'
+ui_cp_charts <- function(country = 'AGO', povline = NULL,
+                         pop_units = 1e6, year_range = 2016:2019,
+                         lkup) {
```
Comment (tonyfujs) [inline R/ui_functions.R:185]: Shouldn't `year_range` be dynamic?

File: R/ui_functions.R (hunk 2 above)
Comment (Aeilert) [reply R/ui_functions.R:289]: Good idea!

File: R/ui_functions.R (hunk 1 above)
Comment (Aeilert) [reply R/ui_functions.R:247]: You are right. Good catch.

File: R/ui_functions.R (hunk 3 above)
Comment (Aeilert) [reply R/ui_functions.R:298]: Also good idea.

File: R/ui_functions.R (hunk 4 above)
Comment (Aeilert) [reply R/ui_functions.R:185]: Do you mean like a parameter in the API?

**Comment by Aeilert** [comment]
> Hey @Aeilert
> Thanks for this. I added a few comments. Take a look and let me know what you think.

Hi @tonyfujs 

Looks good. I addressed your comments. And made a few other minor changes. 

Not sure what was up with the unit test for the number of indicators (intersection between pip() and /indicators), but I counted 20 (and not 19). With polarization now added to indicators_master.csv it is 21. 

All tests and checks pass, so I'm just going to go ahead and merge this.
